### PR TITLE
1. Initializing the Project with basic setup

### PR DESCRIPTION
**Pull request by lor-wtag**
Hello, here is a summary of what is done here:

- Added the required gems
- Established connection with Postgresql
- Created the database

### CODE REVIEW

File: Gemfile
Comment (rsh-wtag) [inline Gemfile:1]: Please sort the gem names lexicographically. Thank you.

File: .dockerignore
Comment (rsh-wtag) [inline .dockerignore:1]: Do we need this file?

File: Dockerfile
Comment (rsh-wtag) [inline Dockerfile:1]: Is this file needed?

File: config/credentials.yml.enc
Comment (rsh-wtag) [inline config/credentials.yml.enc:1]: Please add a new line in this file.

File: .gitignore
Comment (rsh-wtag) [inline .gitignore:1]: Please add a new line at the end of this file.

**Review by rsh-wtag — APPROVED** [review]

File: .dockerignore
Comment (lor-wtag) [reply .dockerignore:1]: Removed the file bhai. Thank you for your suggestion.

File: .gitignore
Comment (lor-wtag) [reply .gitignore:1]: Made the modification bhai. Thank you for your review.

File: config/credentials.yml.enc
Comment (lor-wtag) [reply config/credentials.yml.enc:1]: Updated bhai. Thank you for your review.

File: Dockerfile
Comment (lor-wtag) [reply Dockerfile:1]: Removed the file bhai. Thank you

File: Gemfile
Comment (lor-wtag) [reply Gemfile:1]: Updated bhai. Thank you for your suggestion

**Review by ttu-wtag — APPROVED** [review]

File: test/application_system_test_case.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require "test_helper"
```
Comment (zma-wtag) [inline test/application_system_test_case.rb:1]: can you please remove the whole test folder. as we will be using rspec we don't need to keep this test folder.

File: config/application.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+require "rails/all"
+
+# Require the gems listed in Gemfile, including any gems
+# you've limited to :test, :development, or :production.
+Bundler.require(*Rails.groups)
+
+module OnlineClassroom
+  class Application < Rails::Application
+    # Initialize configuration defaults for originally generated Rails version.
+    config.load_defaults 7.2
```
Comment (zma-wtag) [inline config/application.rb:12]: why we are using this?  `config.load_defaults 7.2`

File: config/credentials.yml.enc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+7nEADfp0Cjg2FNH0rkiOZhvAS4B2r36/DBUDtVUwvgdojeTEWy4HPJAMPPzKi/G1JK0mS5+KRIwonSSOKMXdrZ+GT6qA81DWYzgIkV8NWJbWsfBpByPo2ZfHY+bN8M+1N3IzTQjPsNCKG7C5qMudLi2kcdBD+QPmXu9zLwqhLYpLAum6aVROtn8K+17d0ZOPodyS8dgO6YYldU4Cy/o21hdrJtADU0xBwsZKnfw9sDqrnlLuunwWtqPM8N8vsbD+rLizJjBmmsteCxtpBsnSyfuFmFJfEu3sI/ewRPi8lzH6zhWSTnxQZFDEztfmedfp9VYDz3H2rDOJMpp/9sbFl4+YA+M4wArmrKAuM5YHbS65Qr02vLRWBh+ivTZo5YdFqo3hI6tL1JkZjWX/JxNGkkK0xfF6--wgyx6nCyHBcXgUAg--Vf6Y8nKp1Bf7pq9VGB7uCg==
```
Comment (zma-wtag) [inline config/credentials.yml.enc:1]: purpose of this file?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          # REDIS_URL: redis://localhost:6379/0
+        run: bin/rails db:test:prepare test test:system
+
+      - name: Keep screenshots from failed system tests
+        uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: screenshots
+          path: ${{ github.workspace }}/tmp/screenshots
+          if-no-files-found: ignore
```
Comment (zma-wtag) [inline .github/workflows/ci.yml:101]: lets remove this test running part from this PR. and lets try to introduce rspec running workflow from the mr where we added the rspecs!

Comment (lor-wtag) [reply .github/workflows/ci.yml]: Fixed this bhai. Thank you

File: config/application.rb (hunk 2 above)
Comment (lor-wtag) [reply config/application.rb:12]: Bhai, I got to learn that `config.load_defaults 7.2` loads the default settings for Rails version 7.2, ensuring that the application is set up according to the best practices and configurations recommended for this specific version.
Thank you

File: config/credentials.yml.enc (hunk 3 above)
Comment (lor-wtag) [reply config/credentials.yml.enc:1]: Bhai, `credentials.yml.enc `is to securely store sensitive information in an encrypted format. This way, the information is protected from unauthorized access, and only those with the decryption key can view or use it.

File: test/application_system_test_case.rb (hunk 1 above)
Comment (lor-wtag) [reply test/application_system_test_case.rb:1]: Removed bhai. Thank you

File: Gemfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
+source "https://rubygems.org"
+
+gem "bcrypt", "~> 3.1.7"
+gem "bootsnap", require: false
+gem "cancancan"
```
Comment (evk-wtag) [inline Gemfile:5]: Let's remove this gem from here. We will add the gem in the PR where it is implemented

File: Gemfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
+source "https://rubygems.org"
+
+gem "bcrypt", "~> 3.1.7"
```
Comment (evk-wtag) [inline Gemfile:3]: https://github.com/lor-wtag/Online-Classroom/pull/1/files#r1818030364

File: Gemfile [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  gem "rubocop-rails-omakase", require: false
+end
+
+group :development do
+  gem "web-console"
+end
+
+group :test do
+  gem "capybara"
+  gem "selenium-webdriver"
```
Comment (evk-wtag) [inline Gemfile:30]: Do we need these gems for test purposes? If not, let's remove them

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title><%= content_for(:title) || "Online Classroom" %></title>
```
Comment (evk-wtag) [inline app/views/layouts/application.html.erb:4]: Let's add the i18n gem and move the strings there

File: Gemfile (hunk 7 above)
Comment (lor-wtag) [reply Gemfile]: Updated apu. Removed the gems not required for this PR. Thank you.

File: app/views/layouts/application.html.erb (hunk 8 above)
Comment (lor-wtag) [reply app/views/layouts/application.html.erb]: Made the changes apu. Thank you!

File: test/application_system_test_case.rb (hunk 1 above)
Comment (evk-wtag) [reply test/application_system_test_case.rb:1]: Please address this review ⬆️

Comment (lor-wtag) [reply test/application_system_test_case.rb]: Sorry for the inconvenience apu and bhai. Fixed this now. Thank you

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  #         # REDIS_URL: redis://localhost:6379/0
+  #       run: bin/rails db:test:prepare test test:system
+
+  #     - name: Keep screenshots from failed system tests
+  #       uses: actions/upload-artifact@v4
+  #       if: failure()
+  #       with:
+  #         name: screenshots
+  #         path: ${{ github.workspace }}/tmp/screenshots
+  #         if-no-files-found: ignore
```
Comment (zma-wtag) [inline .github/workflows/ci.yml:101]: remove this from this and introduce rspec from the place where we initiated the Rspec

**Review by zma-wtag — APPROVED** [review]
Good work!

File: .github/workflows/ci.yml (hunk 9 above)
Comment (lor-wtag) [reply .github/workflows/ci.yml]: Fixed bhai. Thank you!

**Review by evk-wtag — APPROVED** [review]
